terraform/modules/networking/subnet.tf

```diff
@@ -0,0 +1,49 @@
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+locals {
+  nat_gateway_cidr_10 = cidrsubnet(aws_vpc.main.cidr_block, 6, 0) # 0.0/22 - public
```

Comment:
Is there a logic to the IP range carving here? 

I think it looks broadly sensible (i.e. there's way more IPs that I expect to need for this type of architecture), but I wonder if it's worth being extra clear in an ADR in case anything changes in future in a way that needs a bunch more IPs.

Reply:
Only that it's harder to accidentally specify overlapping CIDR ranges when they're the same size - and 64 subnets gives us >20 per AZ if we decided to deploy to all 3, which felt like at least double what we'd actually need. But it's pretty arbitrary, happy to go with a different split if I've missed something.

Reply:
I don't think this is worrying, just interested in whether there's a logic to it or it's just a default split, there's a difference between actively thought about decisions that result in a boring answer, vs. just defaulting to something without thinking about it. 

As with a few places I'd like an ADR for this - if you've got too much on, lmk and I'll see if I can pick up some ADRs off the back of this.